uaclient/util.py

```diff
@@ -701,9 +701,9 @@ def write_file(filename: str, content: str, mode: int = 0o644) -> None:
     @param filename: The full path of the file to write.
     @param content: The content to write to the file.
     @param mode: The filesystem mode to set on the file.
-    @param omode: The open mode used when opening the file (w, wb, a, etc.)
     """
     logging.debug("Writing file: %s", filename)
+    content = redact_sensitive_logs(content)
```

Comment:
I think this is a good approach, since we will guarantee that we will always redact sensitive information.
But let's verify with the rest of the team @renanrodrigo @orndorffgrant @blackboxsw

Reply:
This is a good idea, but I think it is too risky.

We often need to write files that _do_ contain secrets (apt auth files and machine-token.json at least), and they may happen to match one of the redaction-regexes. In that case, we could fail to authenticate to an apt repo because we accidentally wrote "<REDACTED>" as the password.

I think to avoid this possibility, we should call `redact_sensitive_logs` over in `action_collect_logs` rather than here for this PR.